apps/web/src/styles/performance.css

```diff
@@ -0,0 +1,16 @@
+/**
+ * Performance optimizations for reducing CLS and improving rendering
+ */
+
+/* Improve rendering performance for off-screen content */
+.location-grid-item {
+  content-visibility: auto;
+  contain-intrinsic-size: auto 300px;
```

Comment:
I don't know the nuance around `contain-intrinsic-size` / `content-visibility`, but this seemed reasonable from skimming MDN:

https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/contain-intrinsic-size